### ISSUE TITLE
"Use module export list" suggestion

### ISSUE THREAD

**Issue by ivan-m** [ndmitchell/hlint#343]
If I have a module that I want to export everything from:

```haskell
module Foo where

data Bar = ...

baz :: Bar -> Bar
baz = ...
```

then hlint will suggest:

```
Warning: Use module export list
Found:
  module Foo where
Why not:
  module Foo (module Foo) where
Note: An explicit list is usually better
```

Is this _really_ a better option?  I didn't actually expect it to work (due to its circular nature) but surprisingly GHC seems to accept it; it still seems rather odd to suggest though.

**Comment by ndmitchell** [comment]
Most people who write a module without an explicit export list should really be including one - there's loads of junk that gets exported from many modules because export lists aren't required. Think of it more like a warning - you should put an explicit export list, sometimes that isn't a good idea, and you've made a conscious decision, in which case adding the explicit export of everything makes that clear. Does that make sense?

**Comment by ivan-m** [comment]
I suppose so.

I still find that `module Foo (module Foo) where` works rather amazing though.
